python/tk_multi_publish2/api/tree.py

```diff
@@ -0,0 +1,284 @@
+# Copyright (c) 2018 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import traceback
+
+import json
+
+import sgtk
+from .item import PublishItem
+
+logger = sgtk.platform.get_logger(__name__)
+
+
+class PublishTree(object):
+    """
+    This class provides an interface for operating on a tree of items to be
+    published. At a high level, a publish tree might be structured like this::
+
+        * root
+            * item1
+                * itemA
+                * itemB
+            * item2
+                * itemC
+            * item3
+                * itemD
+            ...
+
+    Each item in the tree (excluding the root) can have associated tasks.
+
+    Instances of this class are iterable, making traversal very easy:
+
+    .. code-block: python
+
+        for item in publish_tree:
+            # process the item...
+
+    The special, ``root_item`` is exposed as a property on publish tree
```

Comment:
As discussed with Rob, documenting that the `root_item` can be used to store global properties since it's just a regular publish item in the tree. Example below shows how it could be used.